data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/LocationData.java

```diff
@@ -1,43 +1,164 @@
-//TODO Find the best way to attach the location data to a document, and modify this class to allow that.
-
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
 
 public class LocationData {
+    private final String CITY_FIELD = "geo.city_name";
+    private final String COUNTRY_FIELD = "geo.country_name";
+    private final String CONTINENT_FIELD = "geo.continent_code";
+    private final String COUNTRY_ISO_FIELD = "geo.country_iso_code";
+    private final String POSTAL_FIELD = "geo.postal_code";
+    private final String REGION_FIELD = "geo.region_name";
+    private final String REGION_CODE_FIELD = "geo.region_code";
+    private final String TIMEZONE_FIELD = "geo.timezone";
+    private final String LOCATION_FIELD = "geo.location";
+    private final String LATITUDE_FIELD = "geo.location.lat";
+    private final String LONGITUDE_FIELD = "geo.location.lon";
+    private final String ip;
+
+    @JsonProperty(COUNTRY_FIELD)
     private final String countryName;
-    private final String subdivisionName;
+    @JsonProperty(REGION_FIELD)
+    private final String regionName;
+    @JsonProperty(CITY_FIELD)
     private final String cityName;
+    @JsonProperty(LOCATION_FIELD)
+    private final Map<String, Double> location;
+    @JsonProperty(LATITUDE_FIELD)
+    private final Double latitude;
+    @JsonProperty(LONGITUDE_FIELD)
+    private final Double longitude;
+    @JsonProperty(CONTINENT_FIELD)
+    private final String continentCode;
+    @JsonProperty(COUNTRY_ISO_FIELD)
+    private final String countryCode;
+    @JsonProperty(POSTAL_FIELD)
+    private final String postalCode;
+    @JsonProperty(REGION_CODE_FIELD)
+    private final String regionCode;
+    @JsonProperty(TIMEZONE_FIELD)
+    private final String timeZone;
 
-    public LocationData(String countryName, String subdivisionName, String cityName) {
-        this.countryName = countryName;
-        this.subdivisionName = subdivisionName;
-        this.cityName = cityName;
-    }
+    public LocationData(Builder builder) {
+        ip = builder.ip;
+        countryName = builder.countryName;
+        regionName = builder.regionName;
+        cityName = builder.cityName;
+        Double[] latAndLong = builder.location;
+        continentCode = builder.continentCode;
+        countryCode = builder.countryCode;
+        postalCode = builder.postalCode;
+        regionCode = builder.regionCode;
+        timeZone = builder.timeZone;
 
-    @Override
-    public String toString() {
-        return "LocationData{" +
-                "countryName='" + countryName + '\'' +
-                ", subdivisionName='" + subdivisionName + '\'' +
-                ", cityName='" + cityName + '\'' +
-                '}';
+        if (latAndLong != null && latAndLong.length == 2) {
+            location = new HashMap<String, Double>() {{
+                put("lat", latAndLong[0]);
+                put("lon", latAndLong[1]);
+            }};
```

Comment:
Just curious, what's the value of having separate latitude/longitude fields _and_ this one?